src/client/features/admin/index.js

```diff
@@ -174,4 +175,195 @@ class Admin extends React.Component {
   }
 }
 
+class Admin extends React.Component {
+  
+    constructor(props) {
+      super(props);
+  
+      const query = queryString.parse(props.location.search);
+  
+      this.state = {
+        query: _.assign({q: '', gt: 2, lt: 250, type: 'Pathway'}, query),
+        searchResults: [],
+        editLinks: [],
+        dataSources: []
+      };
+  
+      PathwayCommonsService.datasources()
+      .then(result => this.setState({
+        dataSources: result
+        })
+      );
+      this.updateEditLinks = this.updateEditLinks.bind(this);
```

Comment:
Without this piece, the updateEditLinks function uses the wrong scope of `this`.

Reply:
You have a problem elsewhere then.  Remove this line and wherever you get an exception is where the problem is.

Reply:
Functions like updateEditLinks that are called from elsewhere or event handlers are assigned the wrong context by default. React doesn't handle auto rebinding these since ES6 so this is the cleanest way to re-bind them.